src/browser_action/on-each-interaction.js

```diff
@@ -11,43 +11,139 @@
+          group.processingEnd = group.processingEnd ? Math.max(entry.processingEnd, group.processingEnd) : entry.processingEnd;
+        }
     }
+
+    // For the breakdowns we need to know the first and last interaction events (i.e. those wiht interactionIds) in that longest list
+    const longestFrameInteractionEntries = longestFrameEntries.filter((entry) => entry.interactionId);
+    // Entries should be in order so can use that to find first and last
+    const firstInteractionEntry = longestFrameInteractionEntries[0];
+    const lastInteractionEntry = longestFrameInteractionEntries.slice(-1)[0];
+
+    // Filter further to get the entries entry (the longest interactionId events wiht the longest durations)
+    const longestInteractionEntries = longestFrameInteractionEntries.filter((entry) => entry.interactionId  === longestEntry.interactionId && entry.duration === largestDuration);
+
+    // Sometimes target is not set so look across all events with that interactionId to get it
+    const firstInteractionEntryWithTarget = interactions.find((entry) => longestEntry.interactionId === longestEntry.interactionId && entry.target);
+
+    // Filter down LoAFs to ones that interested any event startTime and any processingEnd
+    // Theb LoAF processing the last script in that frame will then present that frame so
+    // we get rAF and other rendering work too with this intrsection.
+    const longAnimationFrameEntries = getIntersectingLoAFs(group.startTime, group.processingEnd)
+
+    // Since entry durations are rounded to the nearest 8ms, we need to clamp
+    // the `nextPaintTime` value to be higher than the `group.processingEnd` or
+    // end time of any LoAF entry.
+    const nextPaintTimeCandidates = [
+      firstInteractionEntry.startTime + firstInteractionEntry.duration,
+      group.processingEnd,
+    ].concat(
+      longAnimationFrameEntries.map((loaf) => loaf.startTime + loaf.duration),
+    );
+    const nextPaintTime = Math.max.apply(Math, nextPaintTimeCandidates);
+
+    // Gather that all in a similar structure to an INP attribution object and send it back
+    callback({
+      name: 'Interaction',
+      value: largestDuration,
+      rating: valueToRating(largestDuration),
+      entries: longestInteractionEntries,
+      attribution: {
+        inputDelay: firstInteractionEntry.processingStart - firstInteractionEntry.startTime,
+        interactionTarget: firstInteractionEntryWithTarget?.target,
+        interactionTime: firstInteractionEntry.startTime,
+        interactionType: firstInteractionEntryWithTarget?.name?.startsWith('key') ? 'keyboard' : 'pointer',
+        longAnimationFrameEntries: longAnimationFrameEntries,
+        nextPaintTime: nextPaintTime,
+        presentationDelay: nextPaintTime - lastInteractionEntry.processingEnd,
+        processedEventEntries: longestFrameEntries,
+        processingDuration: lastInteractionEntry.processingEnd - firstInteractionEntry.processingStart,
```

Comment:
In web-vitals v4, processing isn't just limited to events with `interactionId` values, e.g. a pointer interaction could have a long `mousedown` event that could contribute to its overall processing duration.